hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/DefaultSource.scala

```diff
@@ -213,11 +214,27 @@ class DefaultSource extends RelationProvider
         classOf[HoodieROTablePathFilter],
         classOf[org.apache.hadoop.fs.PathFilter])
 
+      val specifySchema = if (schema == null) {
+        // Load the schema from the commit meta data.
+        // Here we should specify the schema to the latest commit schema since
+        // the table schema evolution.
+        val schemaUtil = new TableSchemaResolver(metaClient)
+        try {
+          Some(SchemaConverters.toSqlType(schemaUtil.getTableAvroSchema)
```

Comment:
Ideally getTableAvroSchema should return Option<Schema> and callers should check if present and proceed accordingly. But in the interest of unblocking others, I am ok with this change.

Reply:
Yes, we can refactor this code in another PR.